mm2src/coins/lightning/ln_events.rs

```diff
@@ -326,10 +343,27 @@ impl LightningEventHandler {
         let payment_preimage = match purpose {
             PaymentPurpose::InvoicePayment { payment_preimage, .. } => match payment_preimage {
                 Some(preimage) => *preimage,
+                // This is a swap related payment since we don't have the preimage yet
                 None => {
-                    // Free the htlc immediately if we don't have the preimage required to claim the payment
-                    // to allow for this inbound liquidity to be used for other inbound payments.
-                    self.channel_manager.fail_htlc_backwards(payment_hash);
+                    let payment_info = PaymentInfo {
+                        payment_hash: *payment_hash,
+                        payment_type: PaymentType::InboundPayment,
+                        description: "Swap Payment".into(),
+                        preimage: None,
+                        secret: None,
+                        amt_msat: Some(received_amount as i64),
+                        fee_paid_msat: None,
+                        status: HTLCStatus::Received,
+                        created_at: (now_ms() / 1000) as i64,
+                        last_updated: (now_ms() / 1000) as i64,
```

Comment:
Well, for `received_amount` to exceed the max of `i64` it will have to be about `92,233,720` BTCs which exceeds the maximum supply of btc.
As for `created_at` and `last_updated` it will be very much in the future, I hope AtomicDEX outlives us until that time :)
But using `TryFrom` for these cases shouldn't hurt, I should use `expect` instead of logging the error.

Reply:
Yeah we don't need to log anything. We can simply use TryFrom/TryInto for those and maybe also create a common function like `try_or_max`